cabbage/testing.py

```diff
@@ -9,37 +9,33 @@
 
 class InMemoryJobStore(store.JobStore):
     def __init__(self):
-        self.jobs: Dict[str, List[jobs.Job]] = {}
+        self.jobs: List[jobs.Job] = []
+        self.finished_jobs: List[Tuple[jobs.Job, jobs.Status]] = []
         self.queues_counter = count()
         self.job_counter = count()
         self.listening_queues = set()
         self.waited = []
-        self.finished_jobs = []
-
-    def register_queue(self, queue: str) -> Optional[int]:
-        self.jobs[queue] = []
-        return next(self.queues_counter)
 
     def launch_job(self, job: jobs.Job) -> int:
         id = next(self.job_counter)
-        self.jobs[job.queue].append(attr.evolve(job, id=id))
+        self.jobs.append(attr.evolve(job, id=id))
 
         return id
 
-    def get_jobs(self, queue: str) -> Iterator[jobs.Job]:
+    def get_jobs(self, queues: Optional[Iterable[str]]) -> Iterator[jobs.Job]:
         # Creating a copy of the iterable so that we can modify it while we iterate
 
-        for job in list(self.jobs[queue]):
-            if not job.scheduled_at or job.scheduled_at <= pendulum.now("UTC"):
-                yield job
+        for job in list(self.jobs):
+            if queues is None or job.queue in queues:
+                if not job.scheduled_at or job.scheduled_at <= pendulum.now("UTC"):
+                    yield job
 
     def finish_job(self, job: jobs.Job, status: jobs.Status) -> None:
-        j = self.jobs[job.queue].pop(0)
-        assert job == j, (job, j)
+        self.jobs.remove(job)
         self.finished_jobs.append((job, status))
 
-    def listen_for_jobs(self, queue: str):
-        self.listening_queues.add(queue)
+    def listen_for_jobs(self, queues: Optional[Iterable[str]] = None) -> None:
+        self.listening_queues.update(queues or ["all"])
```

Comment:
I guess this "all" constant is not used anywhere ? :)

Reply:
Ah yes, sorry, an artifact.